### PR TITLE
restructure plproxy config

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
Move plproxy config into django `DATABASES` setting.

Retake of https://github.com/dimagi/commcare-hq/pull/25802 (without the standby config)

**old**
```python
DATABASES = {}  # normal Django DATABASES setting
config = {
    'shards': {
        'db1': [0, 1],
        'db2': [2, 3],
    },
    'proxy': 'proxy'
}
```
**new**
```python
DATABASES = {
    'proxy': {
        ...
        'PLPROXY': {
            'PROXY': True
        }
    },
    'db1': {
        ...
        'PLPROXY': {
            'SHARDS': [0, 1]  # Shard range for this DB
        }
    },
    'db2': {
        ...
        'PLPROXY': {
            'SHARDS': [1, 2]
        }
    }
}
```

### CODE REVIEW

File: docker/localsettings.py [hunk 1]
```diff
@@ -52,6 +59,9 @@
             'TEST': {
                 'SERIALIZE': False,
             },
+            'PLPROXY': {
+                'SHARDS': [1, 2],
```
Comment (millerdev) [inline docker/localsettings.py]: the old config had `'p2': [2, 3]`. Was the change from `[2, 3]` to `[1, 2]` intentional? The example in the PR description has the same issue.

Comment (snopoke) [reply docker/localsettings.py]: No it wasn't intentional

**Review by emord — APPROVED** [review]
Looks good to me once the documentation is fixed as Daniel mentioned

File: docker/localsettings.py (hunk 1 above)
Comment (snopoke) [inline docker/localsettings.py]:
```suggestion
                'SHARDS': [2, 3],
```

File: corehq/sql_db/config.py [hunk 2]
```diff
@@ -141,16 +101,58 @@ def get_django_shard_map(self):
...
+        else:
+            proxy_db = config_dict['proxy']
+        shard_map = config_dict['shards']
+        host_map = config_dict.get('host_map', {})
+
+        config = PlProxyConfig(proxy_db, shard_map, host_map, _get_shard_count(shard_map.values()))
+        config.validate()
+        return config
+
+    def validate(self):
```
Comment (mkangia) [inline corehq/sql_db/config.py]: should this still be a private method?

**Review by mkangia — COMMENTED** [review]
i see this is already merged, just had couple of comments/queries from code perspective. Feel free to punt them

File: corehq/sql_db/config.py [hunk 3]
```diff
@@ -92,24 +91,22 @@ def num_shards(self):
 
     @property
     def partition_config(self):
-        return settings.PARTITION_DATABASE_CONFIG
+        config = settings.PARTITION_DATABASE_CONFIG
+        if 'groups' in config:
+            # convert old format
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/sql_db/config.py]: Yes, I will add in the warning in a later PR.

File: corehq/sql_db/config.py (hunk 2 above)
Comment (snopoke) [reply corehq/sql_db/config.py]: It could be though I don't thin kit's necessary

**Pull request merged**